server/core/segment.go

```diff
@@ -0,0 +1,25 @@
+package core
+
+import (
+	"database/sql"
+	"log"
+)
+
+// TODO: normally we shouldn't export members of this domain object.
+type Segment struct {
+	Id   int64  `json:"id"`
+	Name string `json:"name"`
+}
+
+func (s *Segment) Save(db *sql.DB) error {
+	err := db.QueryRow(
+		"INSERT INTO segments(name) VALUES($1) RETURNING id",
+		s.Name).Scan(&s.Id)
+
+	if err != nil {
+		log.Fatal("Could not save segment ", s, err)
```

Comment:
`log.Fatal` will exit. I'm not sure if you want to do here. Maybe you do, maybe you don't.